src/roles/actors.rs

```diff
@@ -193,6 +195,29 @@ impl<T: Trait> Roles<T> for Module<T> {
     fn is_role_account(account_id: &T::AccountId) -> bool {
         <ActorByAccountId<T>>::exists(account_id) || <Bondage<T>>::exists(account_id)
     }
+
+    fn account_has_role(account_id: &T::AccountId, role: Role) -> bool {
+        let res = Self::actor_by_account_id(account_id);
+        if res.is_none() {
+            return false;
+        }
+        let actor = res.unwrap();
+        actor.role == role
+    }
+
+    fn random_account_for_role(role: Role) -> Result<T::AccountId, &'static str> {
+        let ids = Self::account_ids_by_role(role);
+        if 0 == ids.len() {
+            return Err(MSG_NO_ACTOR_FOR_ROLE);
+        }
+        let seed = <system::Module<T>>::random_seed();
+        let mut rand: u64 = 0;
+        for offset in 0..8 {
+            rand += (seed.as_ref()[offset] as u64) << offset;
+        }
+        let idx = (rand as usize) % ids.len();
+        return Ok(ids[idx].clone());
```

Comment:
I don't think `clone` is necessary, the value can be moved safely at this point since there is no use of ids after this line. The return statement can also be changed into an expression which will be the return value:

```Ok(ids[idx])```

Reply:
my bad actually, cloning is needed or you get `cannot move out of borrowed content`